### PR TITLE
Added solution to HB05

### PR DESCRIPTION
**Pull request by MichaelShevchenko**

### CODE REVIEW

**Review by floriangeyer21 — APPROVED** [review]

File: src/main/java/com/dev/cinema/service/UserServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+public class UserServiceImpl implements UserService {
+    @Inject
+    private UserDao userDao;
+
+    @Override
+    public User add(User user) {
+        byte[] salt = HashUtil.getSalt();
+        String hashedPassword = HashUtil.hashPassword(user.getPassword(), salt);
+        user.setPassword(hashedPassword);
+        user.setSalt(salt);
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/service/UserServiceImpl.java]:
```suggestion
        user.setSalt(HashUtil.getSalt());
        user.setPassword(HashUtil.hashPassword(user.getPassword(), user.getSalt());
```

File: src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    @Override
+    public User register(String email, String password) {
+        User newUser = new User();
+        newUser.setEmail(email);
+        newUser.setPassword(password);
+        return userService.add(newUser);
+    }
+
+    private boolean isValidPassword(String password, User userFromDbPassword) {
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java]:
```suggestion
    private boolean isPasswordValid(String password, User userFromDbPassword) {
```

File: src/main/java/com/dev/cinema/model/User.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    public void setSalt(byte[] salt) {
+        this.salt = salt;
+    }
+
+    @Override
+    public String toString() {
+        return "User{" + "id=" + id + ", email='" + email + '\''
+                + ", password='" + password + '\'' + '}';
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/model/User.java]: not sure we need password in toString

File: src/main/java/com/dev/cinema/service/UserServiceImpl.java (hunk 1 above)
Comment (MichaelShevchenko) [reply src/main/java/com/dev/cinema/service/UserServiceImpl.java]: done

File: src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java (hunk 2 above)
Comment (MichaelShevchenko) [reply src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java]: done

File: src/main/java/com/dev/cinema/model/User.java (hunk 3 above)
Comment (MichaelShevchenko) [reply src/main/java/com/dev/cinema/model/User.java]: done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**